secant/Features/App/App.swift

```diff
@@ -82,11 +90,77 @@ extension AppReducer {
             
             return Effect(value: .phraseValidation(.displayBackedUpPhrase))
 
+            /// Checking presense of stored wallet in the keychain and presense of database files in documents directory.
+        case .checkWalletInitialisation:
+            var walletInitialisationState: InitialisationState = .uninitialized
+
+            // TODO: Create a dependency to handle database files for the SDK, issue #220 (https://github.com/zcash/secant-ios-wallet/issues/220)
+            let fileManager = FileManager()
+            // TODO: use updated dependency from PR #217 (https://github.com/zcash/secant-ios-wallet/pull/217)
+            let keysPresent = environment.walletStorage.areKeysPresent()
+            
+            do {
+                // TODO: use database URL from the same issue #220
+                let documentsURL = try fileManager.url(for: .documentDirectory, in: .userDomainMask, appropriateFor: nil, create: true)
+                let dataDatabaseURL = documentsURL.appendingPathComponent("ZcashSDK.defaultDataDbName", isDirectory: false)
+                let attributes = try fileManager.attributesOfItem(atPath: dataDatabaseURL.path)
+                let databaseFilesPresent = attributes.isEmpty
+                
+                switch (keysPresent, databaseFilesPresent) {
+                case (false, false):
+                    walletInitialisationState = .uninitialized
+                case (false, true):
+                    walletInitialisationState = .keysMissing
+                case (true, false):
+                    walletInitialisationState = .filesMissing
+                case (true, true):
+                    walletInitialisationState = .initialized
+                }
+            } catch CocoaError.fileNoSuchFile, CocoaError.fileReadNoSuchFile {
+                walletInitialisationState = keysPresent ? .filesMissing : .uninitialized
+            } catch {
+                walletInitialisationState = .failed
+            }
+
+            switch walletInitialisationState {
```

Comment:
Do we need this switch or could we just return these effects right inline?  I think you can remove the intermediate variable up there and just return what you need in the one do block.